scripts/oauth2/README.md

```diff
@@ -0,0 +1,150 @@
+# OAuth2 Test Scripts
+
+This directory contains test scripts for the MCP OAuth2 implementation in Coder.
+
+## Prerequisites
+
+1. Start Coder in development mode:
+
+   ```bash
+   ./scripts/develop.sh
+   ```
+
+2. Login to get a session token:
+
+   ```bash
+   ./scripts/coder-dev.sh login
+   ```
+
+## Scripts
+
+### `test-mcp-oauth2.sh`
+
+Complete automated test suite that verifies all OAuth2 functionality:
+
+- Metadata endpoint
+- PKCE flow
+- Resource parameter support
+- Token refresh
+- Error handling
+
+Usage:
+
+```bash
+chmod +x ./scripts/oauth2/test-mcp-oauth2.sh
+./scripts/oauth2/test-mcp-oauth2.sh
+```
+
+### `setup-test-app.sh`
+
+Creates a test OAuth2 application and outputs environment variables.
+
+Usage:
+
+```bash
+eval $(./scripts/oauth2/setup-test-app.sh)
+echo "Client ID: $CLIENT_ID"
+```
+
+### `cleanup-test-app.sh`
+
+Deletes a test OAuth2 application.
+
+Usage:
+
+```bash
+./scripts/oauth2/cleanup-test-app.sh $CLIENT_ID
+# Or if CLIENT_ID is set as environment variable:
+./scripts/oauth2/cleanup-test-app.sh
+```
+
+### `generate-pkce.sh`
+
+Generates PKCE code verifier and challenge for manual testing.
+
+Usage:
+
+```bash
+./scripts/oauth2/generate-pkce.sh
+```
+
+### `test-manual-flow.sh`
+
+Launches a local Go web server to test the OAuth2 flow interactively. The server automatically handles the OAuth2 callback and token exchange, providing a user-friendly web interface with results.
+
+Usage:
+
+```bash
+# First set up an app
+eval $(./scripts/oauth2/setup-test-app.sh)
+
+# Then run the test server
+./scripts/oauth2/test-manual-flow.sh
+```
+
+Features:
+
+- Starts a local web server on port 9876
+- Automatically captures the authorization code
+- Performs token exchange without manual intervention
+- Displays results in a clean web interface
+- Shows example API calls you can make with the token
```

Comment:
Why are these tests in bash? They will not be run in CI then right?

Reply:
Oh there is a browser component to these scripts?

Reply:
Yeah, it launches a local Go web server that "receives" the authorization code and exchanges it for a coder token after a user clicks on authorize in the browser.